### ISSUE TITLE
XMPP Server monitoring plugin

### ISSUE THREAD

**Issue by BhattiMarry** [nagios-plugins/nagios-plugins#352]
Hello, i was wandering if there is any XMPP server monitoring plugins for the nagios. I am running Ejabberd XMPP server and need a tool/module for its monitoring. There are no open source plugins or modules for monitoring Ejabberd server.

**Comment by BhattiMarry** [comment]
Well that worked solution worked. But this only monitors the ejabberd service, i.e. listening on port 5222. What i am interested in is to monitor different statistics about the server, load, messages in/out, users, memory and all that.

**Comment by JasonRivers** [comment]
I have not found one, but I have started looking in to writing one, I was looking at manipulating the ejabberdctl command to get some of this information, I don't think this shows server load though, for this I'd use check_load or this plugin: https://exchange.nagios.org/directory/Plugins/System-Metrics/CPU-Usage-and-Load/check_proc_cpu-2Esh-3A-check-process-cpu-usage/details which will allow you to get the CPU load of a single process.

As far as I'm aware, ejabberd doesn't expose the number of messages in/out, but you can get the number of users, check a specific user exists E.G. Emailing notification if the "nagios" user doesn't exist on XMPP.

If you're able to point me to some documentation about ejabberd exposing these extra parts I will see what I can do with it.

**Comment by BhattiMarry** [comment]
Yes Ejabberd either does not have any APIs or modules for monitoring as open source OR those modules are outdated far away. The pain in the head is that those modules are written in Erlang, and for me, its like impossible to even understand the written code. However, those modules, if works/integrate, would be everything for Ejabberd monitoring. Take a look please.

- https://github.com/processone/grapherl
- https://github.com/processone/ejabberd-contrib/tree/master/mod_grafite
- https://github.com/processone/ejabberd-contrib/tree/master/mod_statsdx

Thank you in advance for cooperation.

**Comment by JasonRivers** [comment]
Thanks, I'll see what I can find out, However, it won't be an instant solution as I'm working on a few other items at the moment.

**Comment by BhattiMarry** [comment]
No worries. I am also on it.
